ts/main.ts

```diff
@@ -0,0 +1,64 @@
+// interface for Monster
+interface Monster {
+  id: number;
+  name: string;
+  type: string;
+  description: string;
+  locations: [];
+}
+
+const $row = document.querySelector('.row');
+
+async function makeMonsters(): Promise<void> {
+  if (!$row) throw new Error('Could not load row');
+  for (let i = 1; i < 59; i++) {
+    try {
+      const response = await fetch(`https://mhw-db.com/monsters/${i}`);
+      if (!response.ok) {
+        throw new Error('This is not ok');
+      }
+      const monster = (await response.json()) as Monster;
+      console.log(monster);
+      const newPage = renderMonster(monster);
+      $row.append(newPage);
+    } catch (error) {
+      console.log('Error:', error);
+    }
+  }
+}
+makeMonsters();
+
+function renderMonster(object: Monster): HTMLDivElement {
+  /* <div class="row">
+      <div class="column-one-third">
+        <img src='link' class="monster-img"/>
+        <div class="monster-name">
+      </div>
+      </div>
+  */
+
+  object.name = object.name.replace(' ', '-');
+  object.name = object.name.replace("'", '-');
+
```

Comment:
I would conditionally run these. That way we aren't wasting effort when not needed. Also we could assign this to a different variable instead and then we won't run into the earlier issue mentioned of the textContent.

```
let imageUrl = object.name;
if (object.name.includes(' ')) {
  imageUrl = object.name.replace(' ', '-');
}
if (object.name.includes("'") {
  imageUrl = object.name.replace("'", "-");
}
```

These are separate checks because there _could_ be the change of a monster having a dash and an apostrophe.